### DISCUSSION TITLE
Phase 6 in config initialization

### DISCUSSION THREAD

**Discussion by ilvpozzd** [siderolabs/talos#5348] [Q&A] [answered]
What part of config processed in phase 6?
![image](https://user-images.githubusercontent.com/81648305/163419222-9d914350-275c-4a04-8cbd-c0d876e7d971.png)

**Comment by frezbo** [comment, marked as answer]
I believe the whole machineconfig is processed here, are you passing in custom certificates or key material?

**Reply by ilvpozzd** [reply]
Yes, im passing custom certs and keys. 716 byte in yaml config is aggregatorCA's crt (in my case)

**Reply by ilvpozzd** [reply]
What machine/cluster certificates is required for controlplane/worker nodes?

**Reply by frezbo** [reply]
custom certs/keys need to be base64 encoded

**Reply by ilvpozzd** [reply]
Yea, i know it. I ran "talosctl gen config test https://test-addr:6443" and paste generated keys and certs to my Terraform tfvars file. But maybe because I must encode all config to base64 (in vapp variable) it cause some problems in double-decoding?

**Reply by frezbo** [reply]
yeh, seems to be on the TF side, we have an example of using pulumi to generate configs https://github.com/frezbo/pulumi-provider-talos/tree/main/examples

**Reply by ilvpozzd** [reply]
Hmm, maybe because im transform my data through yamlencode() and it add doublequotes to all keys and values in yaml file. Maybe for strings, bools, int it isnt matter but for certs and keys it cause problems with decoding
